packages/flutter/lib/src/animation/animations.dart

```diff
@@ -533,3 +533,58 @@ class TrainHoppingAnimation extends Animation<double>
     return '$currentTrain\u27A9$runtimeType(no next)';
   }
 }
+
+/// A simple interface for combining multiple Animations. Subclasses need only
+/// implement the `value` getter to control how the child animations are
+/// combined. Can be chained to combine more than 2 animations.
+/// Example: to create an animation that is the sum of two others, subclass
+/// this class and define value = first.value + second.value;
+abstract class CompoundAnimation<T> extends Animation<T>
+  with AnimationLazyListenerMixin, AnimationLocalListenersMixin, AnimationLocalStatusListenersMixin {
+  /// Creates a CompoundAnimation. Both arguments must be non-null. Either can
+  /// be a CompoundAnimation itself to combine multiple animations.
+  CompoundAnimation({
+    this.first,
+    this.next,
+  }) {
+    assert(first != null);
+    assert(next != null);
+  }
+
+  /// The first sub-animation. Its status takes precedence if neither are
+  /// animating.
+  final Animation<T> first;
+
+  /// The second sub-animation.
+  final Animation<T> next;
+
+  @override
+  void didStartListening() {
+    first.addListener(notifyListeners);
+    first.addStatusListener(notifyStatusListeners);
+    next.addListener(notifyListeners);
+    next.addStatusListener(notifyStatusListeners);
+  }
+
+  @override
+  void didStopListening() {
+    first.removeListener(notifyListeners);
+    first.removeStatusListener(notifyStatusListeners);
+    next.removeListener(notifyListeners);
+    next.removeStatusListener(notifyStatusListeners);
```

Comment:
this will send redundant status messages. For example, if next.status isn't dismissed, then as first toggles direction we'll keep firing notifications but the status won't have changed.